cmd/check_operator.go

```diff
@@ -57,67 +58,80 @@ var checkOperatorCmd = &cobra.Command{
 			return ErrIndexImageUndefined
 		}
 
-		cfg := runtime.Config{
-			Image:          operatorImage,
-			EnabledChecks:  engine.OperatorPolicy(),
-			ResponseFormat: DefaultOutputFormat,
-			Bundle:         true,
-			Scratch:        true,
-		}
-
-		engine, err := engine.NewForConfig(cfg)
+		cfg, err := runtime.NewConfigFrom(*viper.GetViper())
 		if err != nil {
-			return err
+			return fmt.Errorf("invalid configuration: %w", err)
 		}
 
-		formatter, err := formatters.NewForConfig(cfg)
-		if err != nil {
-			return err
-		}
+		cfg.Image = operatorImage
+		cfg.ResponseFormat = DefaultOutputFormat
+		cfg.Bundle = true
+		cfg.Scratch = true
 
-		// create the results file early to catch cases where we are not
-		// able to write to the filesystem before we attempt to execute checks.
-		resultsFile, err := os.OpenFile(
-			filepath.Join(artifacts.Path(), resultsFilenameWithExtension(formatter.FileExtension())),
-			os.O_WRONLY|os.O_CREATE|os.O_TRUNC,
-			0o600,
-		)
-		if err != nil {
-			return err
-		}
-
-		// also write to stdout
-		resultsOutputTarget := io.MultiWriter(os.Stdout, resultsFile)
-
-		// At this point, we would no longer want usage information printed out
-		// on error, so it doesn't contaminate the output.
+		// Run the operator check
 		cmd.SilenceUsage = true
```

Comment:
See the commends in check_container.go about this.

Reply:
Continuing this discussion in a previous thread.